### ISSUE TITLE
TypeError: Cannot read property 'readyState' of undefined

### ISSUE THREAD

**Issue by Lilja** [websockets/ws#1828]
#### Description
I searched through the other issues with this type of error and I didn't find a solution. I then tried another thing and it worked. So this is merely a report of what I experienced. 

```javascript
var WebSocketServer = require('ws').Server, wss = new WebSocketServer({ server, path: "/ws" });

function runPing(cb) {
   cb('Hello')
}

wss.on('connection', function connection(ws)
  {
    runPing(ws.send)
    ws.on('message', function incoming(message)
      {
        console.log('received: %s', message);
      });

    ws.send('something');
  });
```

```
/home/lilja/code/router-scrape/node_modules/ws/lib/websocket.js:333
    if (this.readyState === WebSocket.CONNECTING) {
             ^

TypeError: Cannot read property 'readyState' of undefined
    at send (/home/lilja/code/router-scrape/node_modules/ws/lib/websocket.js:333:14)
    at runPing (/home/lilja/code/router-scrape/server.js:74:3)
    at WebSocketServer.connection (/home/lilja/code/router-scrape/server.js:93:5)
    at WebSocketServer.emit (events.js:315:20)
    at WebSocketServer.completeUpgrade (/home/lilja/code/router-scrape/node_modules/ws/lib/websocket-server.js:330:5)
    at WebSocketServer.handleUpgrade (/home/lilja/code/router-scrape/node_modules/ws/lib/websocket-server.js:246:10)
    at Server.upgrade (/home/lilja/code/router-scrape/node_modules/ws/lib/websocket-server.js:92:16)
    at Server.emit (events.js:315:20)
    at onParserExecuteCommon (_http_server.js:642:14)
    at onParserExecute (_http_server.js:575:3)

```

But if I change in `runPing` to instead pass the whole websocket param, `ws`:

`runPing(ws.send)` -> `runPing(ws)` and change the references inside `runPing` from `cb` to `ws` and use `ws.send` instead of `cb`. Everything works.

TL;DR: Passing `ws.send` instead of `ws` to a function seems to give some weird error. 

#### Reproducible in:

- version: ws@7.4.1
- Node.js version(s): v13.14.0
- OS version(s): Ubuntu 18.04.5 LTS

**Comment by lpinca** [comment]
This is expected. This is how JavaScript works see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this. In your example you can use

```js
runPing(ws.send.bind(ws));
```
